pkg/component/autoscaling/hvpa/crd.go

```diff
@@ -8,28 +8,82 @@ import (
 	"context"
 	_ "embed"
 
+	"golang.org/x/exp/maps"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/gardener/gardener/pkg/client/kubernetes"
 	"github.com/gardener/gardener/pkg/component"
+	"github.com/gardener/gardener/pkg/utils/flow"
+	kubernetesutils "github.com/gardener/gardener/pkg/utils/kubernetes"
 )
 
-//go:embed templates/crd-autoscaling.k8s.io_hvpas.yaml
-var crd string
+var (
+	//go:embed templates/crd-autoscaling.k8s.io_hvpas.yaml
+	crdHvpas string
+
+	// resourceObjectKeyMap maps the ObjectKey of the CRD to its corresponding manifest
+	resourceObjectKeyMap map[client.ObjectKey]string
```

Comment:
A slice should be fine here? I don't see a benefit using a map here, and the need to copy the crd content.

Reply:
In this example. actually a simple strong variable would suffice, but I thought aligning it with the other implementations prevents confusion.
Should I revert it back to its original form?